libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -1066,8 +1068,39 @@ class abi_serializer_cache_builder {
       }
    }
 
-   std::function<std::optional<abi_serializer>(const account_name& name)> resolver_;
+   resolver_fn_t resolver_;
    abi_serializer_cache_t abi_serializers;
 };
 
+/*
+ * This is equivalent to a resolver, except that everytime the abi_serializer for an account 
+ * is retrieved, it is stored in an unordered_map, so we won't waste time retrieving it again.
+ * This is handy when parsing packed_transactions received in a fc::variant.
+ */
+class caching_resolver {
+public:
+   explicit caching_resolver(resolver_fn_t resolver) :
+      resolver_(std::move(resolver))
+   {
+   }
+
+   std::optional<std::reference_wrapper<const abi_serializer>> operator()(const account_name& account) const {
+      auto it = abi_serializers.find(account);
+      if (it != abi_serializers.end() && it->second)
+         return *it->second;
+      try {
+         auto& dest = abi_serializers[account];
+         dest = resolver_(account);
+         return *dest;
+      } catch( ... ) {
+         throw; // throw if embedded resolver throws
```

Comment:
Why catch and throw here?

Reply:
The catch-throw is still here.

Reply:
Excellent point, it is unneeded. Removed it!